### PR TITLE
add support for Case/When expressions

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

**Review by Jibola — APPROVED** [review]
LGTM

File: django_mongodb/expressions.py [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+from decimal import Decimal
+
+from bson import Decimal128
+from django.core.exceptions import EmptyResultSet, FullResultSet
+from django.db.models.expressions import Case, Col, ExpressionWrapper, Value, When
+
+
+def case(self, compiler, connection):
+    case_parts = []
+    default_mql = self.default.as_mql(compiler, connection)
```
Comment (WaVEV) [inline django_mongodb/expressions.py:11]: I think this one could be in the else part of the for

**Review by WaVEV — APPROVED** [review]
LGTM, just a minor thing.

File: django_mongodb/expressions.py (hunk 1 above)
Comment (timgraham) [reply django_mongodb/expressions.py:11]: I think you're right. I followed the pattern in `Case.as_sql()` so let's see if this change is accepted upstream: https://github.com/django/django/pull/18296.

**Pull request merged**